src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -586,6 +586,55 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     );
   };
 
+  createPendingBlock = async (previousBlock: Block) => {
+    const { miner: minerOptions, chain: chainOptions } = this.#options;
+    const nextBlock = this.#readyNextBlock(previousBlock);
+    // If the executables has inProgress transactions, we were already
+    // in the middle of mining a block when the pending block was requested.
+    // The block that is currently being mined is what we want to return as pending,
+    // so we'll reset the executables to be what it was when mining started,
+    // i.e. all inProgress executables are moved to the front of the pending executables.
+    const executables =
+      this.transactions.transactionPool.cloneAndResetExecutables();
+
+    // the number of transactions that should be included in this block depends
+    // on a few factors. in "instamine" mode (meaning blockTime = 0), we usually
+    // only mine 1 transaction per block. However, if the miner is currently
+    // paused (from miner_stop being called), whenever they restart the miner
+    // all executable transactions will be mined, so the pending block should
+    // contain all executables that will fit in a block. if we're not in
+    // instamine mode, the miner will always try to fill the block, regardless
+    // of whether the miner is paused or not
+    const instamine = this.#instamine;
+    const paused = this.#isPaused();
+    const maxTransactions =
+      instamine && !paused ? Capacity.Single : Capacity.FillBlock;
+
+    // we don't want any events from mining to be
+    // caught by `this.vm`, so we need to make a new one.
+    const vm = await this.createVmFromStateTrie(
+      this.trie.copy(false),
+      chainOptions.allowUnlimitedContractSize,
+      true
+    );
+    // checkpoint to prevent writing any changes to the database
+    await vm.stateManager.checkpoint();
+    const miner = new Miner(
+      minerOptions,
+      executables,
+      vm,
+      this.#readyNextBlock
+    );
+    miner.mine(nextBlock, maxTransactions, true);
+
+    const { block } = await miner.once("block");
+    block.hash = () => {
+      return Quantity.from(null, true);
+    };
+    block.header.stateRoot = Data.from("0x", 32);
```

Comment:
I've created an issue for this: https://github.com/trufflesuite/ganache/issues/3542. I'm not adding a TODO here, because it doesn't seem like we're 100% set on doing this, and I'd rather not have the TODO sticking around forever. If someone chooses to work this issue, it should be easy enough to find this occurence.